worker/mails/mail.go

```diff
@@ -178,7 +199,11 @@ func doSendMail(ctx *job.TaskContext, opts *mail.Options, domain string) error {
 	email := gomail.NewMessage()
 	dialerOptions := opts.Dialer
 	if dialerOptions == nil {
-		dialerOptions = config.GetConfig().Mail
+		if opts.Mode == mail.ModeCampaign {
+			dialerOptions = config.GetConfig().CampaignMail
```

Comment:
IIUC CampaignMail config, unlike Mail, have no defaults (because no flags with default value have been defined) config so we cannot directly use it here without verifying it has been configured.

If it is not defines, I think (correct me if I'm wrong) that Host will have value "" (empty string) which should fail to deliver mail